### DISCUSSION TITLE
can we run pnpm install in parallel for multiple projects using shared pnpm-store

### DISCUSSION THREAD

**Discussion by vasuneet** [pnpm/pnpm#6278] [Q&A]
This query is related to check if we can have shared pnpm store location per node in our CI/CD pipelines. I am pushing this in our company

**Comment by zkochan** [comment]
Yes, it should work. Operations in the store are atomic. You should be able to use install in parallel. But you can't run `pnpm store prune` in parallel to install.

**Reply by Enrice** [reply]
That's a pity. Because for now I HAVE to do exactly that because of #10120!

**Reply by Enrice** [reply]
Specifically I get such errors (seemingly because of concurrency issues):
```
> pnpm store prune && pnpm -r up "xxxx*"

Removed all cached metadata files
 ENOENT  ENOENT: no such file or directory, unlink '/data/pnpm/store/v10/files/00/00cc06fe80fe136b07a2bcc0829446f57762adb43bc05d667ada3fcd8e4bbece2847684cfea95a01e30db9c3868cc3c106c118513973c630b7a64aeec274c7'
```
